### PR TITLE
@intern challenges/two

### PR DESCRIPTION
**Pull request by Macbeth98**
Challenge 2: Uniswap swap Events

### CODE REVIEW

File: uniswap-bot/src/uniswapABI.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'token1',
+    outputs: [{ internalType: 'address', name: '', type: 'address' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
```
Comment (0xtaf) [inline uniswap-bot/src/uniswapABI.ts]: this is not wrong but can also be written like this:

```
export const uniswapPoolABI = [
  "function token0() external view returns (address)",
  "function token1() external view returns (address)",
  "function fee() external view returns (uint24)",
];
```

File: uniswap-bot/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  return poolAddress;
+};
+
+export const isUniswapPoolAddress = async (
+  factoryAddress: string,
+  poolAddress: string,
+  provider: Provider,
+  blockNumber: number
+) => {
+  if (poolAddressCache.get(poolAddress)) return poolAddressCache.get(poolAddress);
```
Comment (0xtaf) [inline uniswap-bot/src/utils.ts:33]:
```suggestion
  if (poolAddressCache.has(poolAddress)) return poolAddressCache.get(poolAddress);
```
using `has` makes more sense

**Review by 0xtaf — CHANGES_REQUESTED** [review]
great job 👏 I don't have any logic related comments

format and `.prettierrc` is missing https://www.notion.so/nethermind/Definition-of-Done-Guidelines-6d81308092214ec6a0d13089127c87f9?pvs=4#ad466074100f443e9cc8cfc95d9e685d

File: uniswap-bot/src/uniswapABI.ts (hunk 1 above)
Comment (Macbeth98) [reply uniswap-bot/src/uniswapABI.ts]: Yes, I just copied the ABI from the etherscan explorer and removed the unused methods. But this is more clean and readable, I will update it.

File: uniswap-bot/src/utils.ts (hunk 2 above)
Comment (Macbeth98) [reply uniswap-bot/src/utils.ts:33]: Yes, I will make changes. 
I forgot to add the .prettierrc file. Will update

Comment (Macbeth98) [reply uniswap-bot/src/utils.ts:33]: Done added the changes! I believe that I can Merge and Close this pull request.

**Pull request merged**